### ISSUE TITLE
Breaking change from 0.14.1 to 0.15?

### ISSUE THREAD

**Issue by DeKi90** [mtkennerly/poetry-dynamic-versioning#79]
Hey, I love this project and I am using it in a bunch of pipelines (azure). Today I realized that poetry-dynamic-versioning no longer manages to determine the version. Locally I am on 0.14.1 and it ran fine, the pipeline ran with 0.17.0 today and failed with this error:

```Error: Unable to determine a dynamic version```

Installed with pip and run as standalone command:
```
python -m pip install --upgrade poetry poetry-dynamic-versioning
poetry-dynamic-versioning
```

pyproject.toml
```
[tool.poetry-dynamic-versioning]
vcs = "git"
style= "pep440"
format-jinja = """
    {%- if distance == 0 -%}
        {{ serialize_pep440(base, stage, revision, metadata=['dirty'] if dirty else None) }}
    {%- elif revision is not none -%}
        {{ serialize_pep440(base, stage, revision + 1, dev=distance, metadata=['dirty'] if dirty else None) }}
    {%- else -%}
        {{ serialize_pep440(bump_version(base), stage, revision, dev=distance, metadata=['dirty'] if dirty else None) }}
    {%- endif -%}
"""
```

Pining the version to ```poetry-dynamic-versioning<0.15``` succeeds again.
I'd be happy for any pointers do investigate the issue further.

**Comment by mtkennerly** [comment]
Thanks for reporting this! There weren't any intentional behavior changes from 0.14.1 to 0.15, so this is definitely a bug. I think I see why it's happening, but I'll take a more thorough look in the morning. In the meantime, could you please try adding `enable = true` to your pyproject.toml and confirm whether `poetry build` correctly produces an artifact with the dynamic version?

**Comment by DeKi90** [comment]
I can confirm that running it as part of the ```poetry build``` toolchain does indeed produce the right result, so I will just change my configuration to use that for the time being.

**Comment by mtkennerly** [comment]
This is now fixed in poetry-dynamic-versioning v0.17.1 and poetry-dynamic-versioning-plugin v0.3.1.
